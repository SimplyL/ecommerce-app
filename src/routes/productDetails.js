import React from 'react';
import { withRouter } from "react-router";
import PropTypes from 'prop-types';

import ProductDescription from '../components/productItem/productDescription';
import Title from '../components/title';
import { PRODUCT_TYPE } from '../constants/types';

export const ProductDetails = props => {
  const { match: { params }, products } = props;
  const product = products.find(item => params.id === item.id);

  return (
    <div>
      <Title name="Product details" />
      <ProductDescription product={product} />
    </div>
  );
};

ProductDetails.propTypes = {
  match: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  products: PropTypes.arrayOf(PRODUCT_TYPE).isRequired,
}

export default withRouter(ProductDetails);
