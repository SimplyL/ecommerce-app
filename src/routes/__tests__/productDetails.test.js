import React from 'react';
import { shallow } from 'enzyme';
import { ProductDetails } from '../productDetails';
import mockProducts from './__mocks__/products';

describe('Product details page', () => {
  let wrapper;
  const [product] = mockProducts;
  const routerMock = {
    match: {
      params: {
        id: '0'
      },
    },
    location: {},
    history: {},
  }

  beforeEach(() => {
    wrapper = shallow(
      <ProductDetails products={mockProducts} {...routerMock} />
    );
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct product details with provided id', () => {
    expect(wrapper.find('ProductDescription').props().product.description).toEqual(product.description);
  });
});
