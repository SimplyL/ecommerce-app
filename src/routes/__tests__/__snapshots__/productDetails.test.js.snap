// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product details page should render without crashing 1`] = `
<div>
  <Title
    name="Product details"
  />
  <ProductDescription
    product={
      Object {
        "description": "Test description",
        "id": "0",
        "image": "roxy-sugar-banana.jpg",
        "name": "Model",
        "price": 4.99,
      }
    }
  />
</div>
`;
